crash in transformSubject

While working on an issue, I noticed a crash in `transformSubject`. on import.

below you find the stack as well as a modified version of the test I was working on that produces the issue.

panic stack
```
panic: runtime error: index out of range [2] with length 1

goroutine 191 [running]:
github.com/nats-io/nats-server/v2/server.(*transform).transform(0xc000177200, {0xc00054b2c0, 0x1, 0x98f787b5478ed9d5})
	/Users/matthiashanel/repos/nats-server/server/accounts.go:4364 +0x679
github.com/nats-io/nats-server/v2/server.(*transform).transformSubject(0xc0008dc31c, {0xc0007e00c2, 0x3})
	/Users/matthiashanel/repos/nats-server/server/accounts.go:4321 +0x1e5
github.com/nats-io/nats-server/v2/server.(*client).processMsgResults(0xc00053c600, 0x1542fe5, 0xc0000ec180, {0xc0008dc358, 0x4, 0xa8}, {0x0, 0x0, 0x0}, {0xc0008dc308, ...}, ...)
	/Users/matthiashanel/repos/nats-server/server/client.go:4162 +0x7cd
github.com/nats-io/nats-server/v2/server.(*client).processInboundRoutedMsg(0xc00053c600, {0xc0008dc358, 0x0, 0xa8})
	/Users/matthiashanel/repos/nats-server/server/route.go:443 +0x159
github.com/nats-io/nats-server/v2/server.(*client).processInboundMsg(0xc00053c600, {0xc0008dc358, 0x51, 0xfb})
	/Users/matthiashanel/repos/nats-server/server/client.go:3512 +0x36
github.com/nats-io/nats-server/v2/server.(*client).parse(0xc00053c600, {0xc0008dc300, 0x5c, 0xc09280d889ebfe48})
	/Users/matthiashanel/repos/nats-server/server/parser.go:497 +0x246a
github.com/nats-io/nats-server/v2/server.(*client).readLoop(0xc00053c600, {0x0, 0x0, 0x0})
	/Users/matthiashanel/repos/nats-server/server/client.go:1229 +0xe1f
github.com/nats-io/nats-server/v2/server.(*Server).createRoute.func1()
	/Users/matthiashanel/repos/nats-server/server/route.go:1372 +0x25
created by github.com/nats-io/nats-server/v2/server.(*Server).startGoRoutine
	/Users/matthiashanel/repos/nats-server/server/server.go:3013 +0x87

```

Modified unit test that produces the issue:
```
func TestJetStreamSuperClusterImportConsumerStreamSubjectRemap(t *testing.T) {
	template := `
	listen: 127.0.0.1:-1
	server_name: %s
	jetstream: {max_mem_store: 256MB, max_file_store: 2GB, domain: HUB, store_dir: '%s'}

	cluster {
		name: %s
		listen: 127.0.0.1:%d
		routes = [%s]
	}

	accounts: {
		JS: {
			jetstream: enabled
			users: [ {user: js, password: pwd} ]
			exports [
				# This is streaming to a delivery subject for a push based consumer.
				{ stream: "eliver.ORDERS.*" }
				# This is to ack received messages. This is a service to support sync ack.
				{ service: "$JS.ACK.ORDERS.*.>" }
				# To support ordered consumers, flow control.
				{ service: "$JS.FC.>" }
			]
		},
		IM: {
			users: [ {user: im, password: pwd} ]
			imports [
				{ stream:  { account: JS, subject: "eliver.ORDERS.*" }, to: "deliver.ORDERS.*"}

				{ service: {account: JS, subject: "$JS.FC.>" }}
			]
		},
		$SYS { users = [ { user: "admin", pass: "s3cr3t!" } ] },
	}
	leaf {
		listen: 127.0.0.1:-1
	}`

	test := func(t *testing.T, queue bool) {
		c := createJetStreamSuperClusterWithTemplate(t, template, 3, 2)
		defer c.shutdown()

		s := c.randomServer()
		nc, js := jsClientConnect(t, s, nats.UserInfo("js", "pwd"))
		defer nc.Close()

		_, err := js.AddStream(&nats.StreamConfig{
			Name:      "ORDERS",
			Subjects:  []string{"foo"}, // The JS subject.
			Replicas:  3,
			Placement: &nats.Placement{Cluster: "C1"},
		})
		require_NoError(t, err)

		_, err = js.Publish("foo", []byte("OK"))
		require_NoError(t, err)

		for dur, deliver := range map[string]string{
			"dur-route":   "eliver.ORDERS.route",
			"dur-gateway": "eliver.ORDERS.gateway",
			"dur-leaf-1":  "eliver.ORDERS.leaf1",
			"dur-leaf-2":  "eliver.ORDERS.leaf2",
		} {
			cfg := &nats.ConsumerConfig{
				Durable:        dur,
				DeliverSubject: deliver,
				AckPolicy:      nats.AckExplicitPolicy,
			}
			if queue {
				cfg.DeliverGroup = "queue"
			}
			_, err = js.AddConsumer("ORDERS", cfg)
			require_NoError(t, err)
		}

		testCase := func(t *testing.T, s *Server, dSubj string) {
			nc2, err := nats.Connect(s.ClientURL(), nats.UserInfo("im", "pwd"))
			require_NoError(t, err)
			defer nc2.Close()

			var sub *nats.Subscription
			if queue {
				sub, err = nc2.QueueSubscribeSync(dSubj, "queue")
			} else {
				sub, err = nc2.SubscribeSync(dSubj)
			}
			require_NoError(t, err)

			m, err := sub.NextMsg(time.Second)
			require_NoError(t, err)

			if m.Subject != "foo" {
				t.Fatalf("Subject not mapped correctly across account boundary, expected %q got %q", "foo", m.Subject)
			}
			require_False(t, strings.Contains(m.Reply, "@"))
		}

		t.Run("route", func(t *testing.T) {
			// pick random non consumer leader so we receive via route
			s := c.clusterForName("C1").randomNonConsumerLeader("JS", "ORDERS", "dur-route")
			testCase(t, s, "deliver.ORDERS.route")
		})
		t.Run("gateway", func(t *testing.T) {
			// pick server with inbound gateway from consumer leader, so we receive from gateway and have no route in between
			scl := c.clusterForName("C1").consumerLeader("JS", "ORDERS", "dur-gateway")
			var sfound *Server
			for _, s := range c.clusterForName("C2").servers {
				s.mu.Lock()
				for _, c := range s.gateway.in {
					if c.GetName() == scl.info.ID {
						sfound = s
						break
					}
				}
				s.mu.Unlock()
				if sfound != nil {
					break
				}
			}
			testCase(t, sfound, "deliver.ORDERS.gateway")
		})
		t.Run("leaf-post-export", func(t *testing.T) {
			// create leaf node server connected post export/import
			scl := c.clusterForName("C1").consumerLeader("JS", "ORDERS", "dur-leaf-1")
			cf := createConfFile(t, []byte(fmt.Sprintf(`
			port: -1
			leafnodes {
				remotes [ { url: "nats://im:pwd@127.0.0.1:%d" } ]
			}
			authorization: {
				user: im,
				password: pwd
			}
		`, scl.getOpts().LeafNode.Port)))
			defer removeFile(t, cf)
			s, _ := RunServerWithConfig(cf)
			defer s.Shutdown()
			checkLeafNodeConnected(t, scl)
			testCase(t, s, "deliver.ORDERS.leaf1")
		})
		t.Run("leaf-pre-export", func(t *testing.T) {
			// create leaf node server connected pre export, perform export/import on leaf node server
			scl := c.clusterForName("C1").consumerLeader("JS", "ORDERS", "dur-leaf-2")
			cf := createConfFile(t, []byte(fmt.Sprintf(`
			port: -1
			leafnodes {
				remotes [ { url: "nats://js:pwd@127.0.0.1:%d", account: JS2 } ]
			}
			accounts: {
				JS2: {
					users: [ {user: js, password: pwd} ]
					exports [
						# This is streaming to a delivery subject for a push based consumer.
						{ stream: "deliver.ORDERS.leaf2" }
						# This is to ack received messages. This is a service to support sync ack.
						{ service: "$JS.ACK.ORDERS.*.>" }
						# To support ordered consumers, flow control.
						{ service: "$JS.FC.>" }
					]
				},
				IM2: {
					users: [ {user: im, password: pwd} ]
					imports [
						{ stream:  { account: JS2, subject: "deliver.ORDERS.leaf2" }}
						{ service: {account: JS2, subject: "$JS.FC.>" }}
					]
				},
			}
		`, scl.getOpts().LeafNode.Port)))
			defer removeFile(t, cf)
			s, _ := RunServerWithConfig(cf)
			defer s.Shutdown()
			checkLeafNodeConnected(t, scl)
			testCase(t, s, "deliver.ORDERS.leaf2")
		})
	}

	t.Run("noQueue", func(t *testing.T) {
		test(t, false)
	})
	t.Run("queue", func(t *testing.T) {
		test(t, true)
	})
}

```

--- a/server/client.go
+++ b/server/client.go
@@ -2654,7 +2654,7 @@ func (c *client) addShadowSub(sub *subscription, ime *ime) (*subscription, error
 	nsub := *sub // copy
 	nsub.im = im
 
-	if !im.usePub && ime.dyn {
+	if !im.usePub && ime.dyn && im.tr != nil {
 		if im.rtr == nil {
 			im.rtr = im.tr.reverse()
 		}
@@ -4159,7 +4159,7 @@ func (c *client) processMsgResults(acc *Account, r *SublistResult, msg, deliver,
 				continue
 			}
 			if sub.im.tr != nil {
-				to, _ := sub.im.tr.transformSubject(string(dsubj))
+				to, _ := sub.im.tr.transformSubject(string(subject))
 				dsubj = append(_dsubj[:0], to...)
 			} else if sub.im.usePub {
 				dsubj = append(_dsubj[:0], subj...)
@@ -4300,7 +4300,7 @@ func (c *client) processMsgResults(acc *Account, r *SublistResult, msg, deliver,
 					continue
 				}
 				if sub.im.tr != nil {
-					to, _ := sub.im.tr.transformSubject(string(subj))
+					to, _ := sub.im.tr.transformSubject(string(subject))
 					dsubj = append(_dsubj[:0], to...)
 				} else if sub.im.usePub {
 					dsubj = append(_dsubj[:0], subj...)

--- a/server/jetstream_test.go
+++ b/server/jetstream_test.go
@@ -17006,7 +17006,8 @@ func TestJetStreamImportConsumerStreamSubjectRemapSingle(t *testing.T) {
 				users: [ {user: js, password: pwd} ]
 				exports [
 					# This is streaming to a delivery subject for a push based consumer.
-					{ stream: "deliver.ORDERS" }
+					{ stream: "deliver.*" }
+					{ stream: "foo.*" }
 					# This is to ack received messages. This is a service to support sync ack.
 					{ service: "$JS.ACK.ORDERS.*.>" }
 					# To support ordered consumers, flow control.
@@ -17016,8 +17017,9 @@ func TestJetStreamImportConsumerStreamSubjectRemapSingle(t *testing.T) {
 			IM: {
 				users: [ {user: im, password: pwd} ]
 				imports [
-					{ stream:  { account: JS, subject: "deliver.ORDERS" }, to: "d" }
-					{ service: {account: JS, subject: "$JS.FC.>" }}
+					{ stream:  { account: JS, subject: "deliver.ORDERS" }, to: "d.*" }
+					{ stream:  { account: JS, subject: "foo.*" }, to: "bar.*" }
+					{ service: { account: JS, subject: "$JS.FC.>" }}
 				]
 			},
 		}
@@ -17061,14 +17063,36 @@ func TestJetStreamImportConsumerStreamSubjectRemapSingle(t *testing.T) {
 
 		var sub *nats.Subscription
 		if queue {
-			sub, err = nc2.QueueSubscribeSync("d", queueName)
+			sub, err = nc2.QueueSubscribeSync("d.ORDERS", queueName)
 			require_NoError(t, err)
 		} else {
-			sub, err = nc2.SubscribeSync("d")
+			sub, err = nc2.SubscribeSync("d.ORDERS")
 			require_NoError(t, err)
 		}
 
 		m, err := sub.NextMsg(time.Second)
+		require_NoError(t, err)
+
+		if m.Subject != "foo" {
+			t.Fatalf("Subject not mapped correctly across account boundary, expected %q got %q", "foo", m.Subject)
+		}
+
+		// Now do one that would kick in a transform.
+		_, err = js.AddConsumer("ORDERS", &nats.ConsumerConfig{
+			DeliverSubject: "foo.ORDERS",
+			AckPolicy:      nats.AckExplicitPolicy,
+			DeliverGroup:   queueName,
+		})
+		require_NoError(t, err)
+
+		if queue {
+			sub, err = nc2.QueueSubscribeSync("bar.ORDERS", queueName)
+			require_NoError(t, err)
+		} else {
+			sub, err = nc2.SubscribeSync("bar.ORDERS")
+			require_NoError(t, err)
+		}
+		m, err = sub.NextMsg(time.Second)
 		require_NoError(t, err)
 
 		if m.Subject != "foo" {

